content/client-lib-development-guide/features.textile

```diff
@@ -682,18 +693,17 @@ h4. ChannelOptions
 * @(TB1)@ options provided when instancing a channel, see "Java ChannelOptions":https://github.com/ably/ably-java/blob/master/src/io/ably/types/ChannelOptions.java as a reference
 * @(TB2)@ The attributes of @ChannelOptions@ consist of:
 ** @(TB2a)@ @encrypted@ boolean - when true, enables automatic encryption & decryption of all messages
-** @(TB2b)@ @cipherParams@ @CipherParams@ - when encrypted is true, the cipher params are required
+** @(TB2b)@ @cipherParams@ - This is required if @encrypted@ is true. It consists of either:
+*** @(TB2b1)@ A @CipherParams@ instance, or
+*** @(TB2b2)@ an options hash (or language equivalent) consisting of the key @key@ with value a key. In this case, the client library should call `getDefaultParams`, passing it the key, to obtain a @CipherParams@ instance
```

Comment:
> `channels.get("foo", new ChannelOptions(Crypto.getDefaultParams("key")));`

I am not mad on this, but can't quite tell you why.  Perhaps it's simply because it's not obvious why you're passing in that object.

> What I'm proposing is providing this as a convenience
> `channels.get("foo", ChannelOptions.withCipherKey("key"));`

I don't have an issue with that although equally not sure it's really needed. If we did go down that route though, I would prefer to have `ChannelOptions.withCipher("key")` because then it would simply map directly to `getDefaultParams` and also support `ChannelOptions.withCipher({ "key": key, "algorithm": "cbc" })` etc (I know that is not quite valid because they would be `Params[]`, but for simplicity it's written that way).  Also, I am not sure what the conventions would be about naming a constructor method like this i.e. should it be prefixed with the word `new` or `get` perhaps?